### ISSUE TITLE
[BUG] service bus sometimes crashes with ClassCastException

### ISSUE THREAD

**Issue by hamnis** [Azure/azure-sdk-for-java#40610]
**Describe the bug**
Snippet from ServiceBusMessage.java

```
    public OffsetDateTime getScheduledEnqueueTime() {
        Object value = amqpAnnotatedMessage.getMessageAnnotations().get(SCHEDULED_ENQUEUE_UTC_TIME_NAME.getValue());
        return value != null
            ? ((OffsetDateTime) value).toInstant().atOffset(ZoneOffset.UTC)
            : null;
    }
```
The OffsetDateTime cast is not safe.

***Exception or Stack Trace***
```
com.azure.messaging.servicebus.ServiceBusException: class java.util.Date cannot be cast to class java.time.OffsetDateTime (java.util.Date and java.time.OffsetDateTime are in module java.base of loader 'bootstrap')
	at com.azure.messaging.servicebus.ServiceBusSenderAsyncClient.mapError(ServiceBusSenderAsyncClient.java:914)
	at reactor.core.publisher.Mono.lambda$onErrorMap$31(Mono.java:3811)
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:94)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.secondError(MonoFlatMap.java:192)
	at reactor.core.publisher.MonoFlatMap$FlatMapInner.onError(MonoFlatMap.java:259)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onError(MonoFlatMap.java:172)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.secondError(MonoFlatMap.java:192)
	at reactor.core.publisher.MonoFlatMap$FlatMapInner.onError(MonoFlatMap.java:259)
	at reactor.core.publisher.MonoStreamCollector$StreamCollectorSubscriber.onError(MonoStreamCollector.java:150)
	at reactor.core.publisher.MonoStreamCollector$StreamCollectorSubscriber.onNext(MonoStreamCollector.java:137)
	at reactor.core.publisher.Operators$ScalarSubscription.request(Operators.java:2400)
	at reactor.core.publisher.MonoStreamCollector$StreamCollectorSubscriber.onSubscribe(MonoStreamCollector.java:121)
	at reactor.core.publisher.FluxJust.subscribe(FluxJust.java:68)
	at reactor.core.publisher.MonoFromFluxOperator.subscribe(MonoFromFluxOperator.java:81)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:157)
	at reactor.core.publisher.Operators$ScalarSubscription.request(Operators.java:2400)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onSubscribe(MonoFlatMap.java:110)
	at reactor.core.publisher.MonoJust.subscribe(MonoJust.java:55)
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:53)
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:64)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:157)
	at reactor.core.publisher.MonoPeekTerminal$MonoTerminalPeekSubscriber.onNext(MonoPeekTerminal.java:180)
	at reactor.core.publisher.SerializedSubscriber.onNext(SerializedSubscriber.java:99)
	at reactor.core.publisher.FluxRetryWhen$RetryWhenMainSubscriber.onNext(FluxRetryWhen.java:174)
	at reactor.core.publisher.SerializedSubscriber.onNext(SerializedSubscriber.java:99)
	at reactor.core.publisher.FluxTimeout$TimeoutMainSubscriber.onNext(FluxTimeout.java:180)
	at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.onNext(FluxPeekFuseable.java:210)
...
```

**To Reproduce**
Unsure why this happens, but our pod sometimes crashes with this.

***Code Snippet***
this happens in the library.

**Expected behavior**

**Screenshots**

**Setup (please complete the following information):**
 - OS: Kubernetes/Linux
 - IDE: IntelliJ
 - Library/Libraries: "com.azure" % "azure-messaging-servicebus" % "7.17.0",
 - Java version: 21
 - App Server/Environment: Kubernetes
 - Frameworks: cats-effect, Scala
 

**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by lmolkova** [comment]
Thank you for reporting this @hamnis!

Could you please provide a few more details:
- do you know which clients are used to send messages to the Service Bus? Their languages/SDKs/versions?
- is it something that started happening recently, e.g. on a specific version of ServiceBus SDK, or did it happen previously?

Thanks!

**Comment by hamnis** [comment]
> Thank you for reporting this @hamnis!
> 
> Could you please provide a few more details:
> 
>     * do you know which clients are used to send messages to the Service Bus? Their languages/SDKs/versions?

I think it's some f# thing, but I am not sure.

> 
>     * is it something that started happening recently, e.g. on a specific version of ServiceBus SDK, or did it happen previously?
>

We upgrade all libraries every week, and this has been crashing multiple times per day for some time, but with different reasons.

**Comment by lmolkova** [comment]
@hamnis 

I believe I have a fix in the #40642 . Could you please confirm my assumption about your application:
- first schedules a message to be sent to a service bus queue/topic
- then receives the message (in the java service that crashes with cast exception)
- uses this received message to create a new `ServiceBusMessage`
- sends this message to a different servicebus entity

Is this right?

If my assumption is correct, you should be able to workaround this issue (until the fix is released) by removing the value under `SCHEDULED_ENQUEUE_UTC_TIME_NAME` message property with the following code:

```java
ServiceBusMessage messageToSend = new ServiceBusMessage(receivedMessage);
messageToSend .getRawAmqpMessage().getMessageAnnotations().remove(AmqpMessageConstant.SCHEDULED_ENQUEUE_UTC_TIME_NAME.getValue());
```
